src/linker/Linker.Steps/OutputStep.cs

```diff
@@ -76,11 +76,15 @@ static TargetArchitecture CalculateArchitecture (TargetArchitecture readyToRunAr
 			throw new BadImageFormatException ("unrecognized module attributes");
 		}
 
+		protected override bool ConditionToProcess ()
```

Comment:
I think after ProcessWarningsStep, there's no step that could produce any warning/error. Should these steps have this condition as well, or maybe short-circuit steps after this point if errors were found?

Reply:
I don't think it's worth it (optimize for failure), they should not produce errors but they could in the future.